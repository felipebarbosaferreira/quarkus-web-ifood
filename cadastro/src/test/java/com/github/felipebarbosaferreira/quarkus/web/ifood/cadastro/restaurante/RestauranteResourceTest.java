package com.github.felipebarbosaferreira.quarkus.web.ifood.cadastro.restaurante;

import static io.restassured.RestAssured.given;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;
import com.github.felipebarbosaferreira.quarkus.web.ifood.cadastro.PostgresDBLifeCycleControlTest;

import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;

@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTest
@QuarkusTestResource(PostgresDBLifeCycleControlTest.class)
public class RestauranteResourceTest {

	@Test
	@DataSet("test-restaurante-listar-todos-1.yml")
	public void testListarTodosEndpoint() {
		String retorno = given().when().get("/restaurantes").then().statusCode(200).extract().asString();
		Approvals.verifyJson(retorno);
	}

	@Test
	@DataSet("test-restaurante-listar-todos-1.yml")
	public void testObterEndpoint() {
		String retorno = given().when().get("/restaurantes/2").then().statusCode(200).extract().asString();
		Approvals.verifyJson(retorno);
	}

}