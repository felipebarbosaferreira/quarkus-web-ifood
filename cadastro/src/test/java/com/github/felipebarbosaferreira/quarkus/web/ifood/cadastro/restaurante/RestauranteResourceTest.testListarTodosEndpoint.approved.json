[
  {
    "id": 1,
    "nome": "Restaurante UM",
    "proprietario": "Proprietario UM"
  },
  {
    "id": 2,
    "nome": "Restaurante Dois",
    "proprietario": "Proprietario Dois"
  },
  {
    "id": 3,
    "nome": "Restaurante Tres",
    "proprietario": "Proprietario Tres"
  }
]